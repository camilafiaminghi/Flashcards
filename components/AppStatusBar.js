import React from 'react'
import PropTypes from 'prop-types'
import { View, StatusBar } from 'react-native'
import { Constants } from 'expo'

const AppStatusBar = ({ backgroundColor, ...props }) => {
	return (
		<View style={{ backgroundColor, height: Constants.statusBarHeight }}>
			<StatusBar translucent backgroundColor={backgroundColor} {...props} />
		</View>
	)
}

AppStatusBar.propTypes = {
	backgroundColor: PropTypes.string.isRequired
}

export default AppStatusBar
