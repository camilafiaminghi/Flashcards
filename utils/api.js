import { AsyncStorage } from 'react-native'
export const FLASHCARDS_STORAGE_KEY = 'Flashcards:decks'

const obj = {
	'JavaScript' : {
		title: 'JavaScript',
		questions: []
	},
	'React' : {
		title: 'React',
		questions: []
	}
}

export function setInitialData (data=obj) {
	// return AsyncStorage.setItem(FLASHCARDS_STORAGE_KEY, JSON.stringify(data))
	return AsyncStorage.mergeItem(FLASHCARDS_STORAGE_KEY, JSON.stringify(data))
}

export function getDecks () {
	return AsyncStorage.getItem(FLASHCARDS_STORAGE_KEY)
}

export function getDeck (entryId) {
	return AsyncStorage.getItem(FLASHCARDS_STORAGE_KEY)
		.then((results) => {
			const data = JSON.parse(results)
			return data[entryId]
		})
}

export function saveDeckTitle ({ title }) {
	return AsyncStorage.mergeItem(FLASHCARDS_STORAGE_KEY, JSON.stringify({
		[title]: {
			title,
			questions: []
		}
	}))
}

export function addCardToDeck (entryId, card) {
	return getDeck(entryId)
		.then((entry) => {
			return AsyncStorage.mergeItem(FLASHCARDS_STORAGE_KEY, JSON.stringify({
				[entryId]: {
					...entry,
					questions: [...entry.questions, card]
				}
			}))
		})
}
