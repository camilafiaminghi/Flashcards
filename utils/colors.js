// utils/colors.js
export const textColor = '#333333'
export const textColorInverse = '#ffffff'
export const pColor = '#9fa8da'
export const pColorLight = '#d1d9ff'
export const pColorDark = '#6f79a8'
export const sColor = '#bdbdbd'
export const sColorLight = '#efefef'
export const sColorDark = '#8d8d8d'

