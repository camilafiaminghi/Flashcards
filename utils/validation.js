export function validationRules (name, value) {
	const rule = {
		title: !!value.trim() && value.length > 6,
		body: !!value.trim() && value.length > 6
	}

	return rule[name]
}

export function isValid (form) {
	return Object.keys(form).every(function (key) {
		return form[key]
	});
}
