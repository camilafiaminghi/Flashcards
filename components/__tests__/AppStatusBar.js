import React, { Component } from 'react'
import renderer from 'react-test-renderer'
import AppStatusBar from '../AppStatusBar'
import { pColorDark } from '../../utils/colors'

let wrapper
let instance
let component
const props = {
	backgroundColor: pColorDark
}

describe('<AppStatusBar />', () => {

	beforeEach(() => {
		component = shallow(<AppStatusBar {...props} />)
	})

	it('matches snapshot', () => {
    wrapper = renderer.create(<AppStatusBar {...props} />)
    expect(wrapper.toJSON()).toMatchSnapshot()
  });

	it('should render', () => {
		expect(wrapper).toBeTruthy()
	})
})
