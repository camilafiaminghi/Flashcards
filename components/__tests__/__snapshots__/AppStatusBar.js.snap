// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppStatusBar /> matches snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#6f79a8",
      "height": 20,
    }
  }
/>
`;
